packages/headless/src/api/analytics/search-analytics.ts

```diff
@@ -75,6 +75,21 @@ export class SearchAnalyticsProvider
     return hasSplitTestRun ? effectivePipelineWithSplitTestRun : undefined;
   }
 
+  public getBaseMetadata() {
+    const state = this.getState();
+    const baseObject = super.getBaseMetadata();
+
+    const generativeQuestionAnsweringId =
+      state.search?.response?.extendedResults?.generativeQuestionAnsweringId;
+
+    if (generativeQuestionAnsweringId) {
```

Comment:
This whole function seems like a hack because `coveo.analytics` does not accept the proper type ?

Reply:
This is fair 😅. I went with the way we add the headless version in the payload but it is indeed hacky. I'll see about adding it in coveo.analytics.

Reply:
I was asked to set it as custom event data here and it's supposed to be temporary until we have the new UA protocol. I could add a getter to the provider given to the analytics client and set it in there but is it really worth it? For now headless is the only lib that will use this.

Reply:
"Supposed to be temporary" 😉 

Alright then 😄

Reply:
Our UA plans are the thing i'm able to be the least certain about so I don't want to risk being too confident 😅